src/redux-middleware/updateUrlHistory.ts

```diff
@@ -53,7 +52,8 @@ const updateUrlHistory = (state: State, path = RANKED_ROOT, { replace, contextVi
   // update browser history
   try {
     window.history[replace ? 'replaceState' : 'pushState'](
```

Comment:
There is something causing `[ROOT]` to get pushed onto the stack when skipping an invalid history state. This seems to happen on `dev` as well. I would suggest identifying the cause of that, as it might be interfering with the skipping back/forward functionality.